src/Nirum/Parser.hs

```diff
@@ -407,10 +408,15 @@ recordTypeDeclaration = do
     annotationSet'' <- annotationsWithDocs annotationSet' docs'
     return $ TypeDeclaration typename (RecordType fields') annotationSet''
 
-tag :: Parser Tag
+tag :: Parser (Tag, Bool)
 tag = do
     annotationSet' <- annotationSet <?> "union tag annotations"
     spaces
+    -- CHECK: If a new reserved keyword is introduced, it has to be also
+    -- added to `reservedKeywords` set in the `Nirum.Constructs.Identifier`
+    -- module.
```

Comment:
This check comment should go to the top of *Parser.hs* source.  See also [Checkmate docs][1].  Since it's sensitive to block indentation levels, if a `CHECK` comment is too deeply inside it could be hardly warned.

[1]: https://github.com/spoqa/checkmate#listing-relevant-checks-overlapped-blocks